### ISSUE TITLE
[ENH] Unify pretrained/foundation forecasters under a shared base class

### ISSUE THREAD

**Issue by geetu040** [alan-turing-institute/sktime#10498]
Detailed Plan: https://hackmd.io/@geetu/B1Ccj-HXGx

`sktime` now has many pretrained/foundation-model forecasters, including Chronos, TimesFM, TinyTimeMixer, LagLlama, MOIRAI, Timer, Toto, Kronos, WindFM, and generic Hugging Face Transformers models.

These wrappers currently duplicate a lot of infrastructure: model loading, caching, device handling, serialization, prediction formatting, quantile handling, and fine-tuning/pretraining setup. Their APIs also expose device, PEFT, quantization, and training behavior inconsistently.

This issue proposes introducing a shared internal base class for foundation-model forecasters.

**Proposal**

Add an internal `BaseFoundationForecaster` that subclasses `BaseForecaster` and owns the common lifecycle:

- `_fit`, `_pretrain`, `_predict`, `_predict_quantiles`, `_predict_interval`
- model loading and cache lookup
- serialization/reload behavior
- device/dtype/quantization config handling
- normalized forecast result formatting

Concrete estimators should subclass it directly and implement only model-specific hooks, such as:

```python
class BaseFoundationForecaster(BaseForecaster):
    def _get_model_load_config(self): ...
    def _load_model(self, load_config, tune_config): ...
    def _make_context(self, y, X, fh, cutoff, handle): ...
    def _predict_native(self, handle, context, request): ...
```

Supporting internal objects could include:

`ModelLoadConfig`, `FineTuneConfig`, `ParameterEfficientTuneConfig`, `InferenceConfig`, `FoundationModelSpec`, `ModelHandle`, `ModelContext`, `ForecastRequest`, `ForecastResult`

**Acceptance Criteria**

- A shared internal `BaseFoundationForecaster` exists.
- Common load/cache/device/serialization/prediction-formatting logic is implemented once.
- At least 2-3 representative foundation forecasters are migrated.
- Existing public estimator constructors remain backward compatible.
- Quantization, PEFT, pretraining, zero-shot inference, and probabilistic outputs have normalized internal config/result objects.
- Added test coverage.
- Migrated estimators keep their existing user-facing behavior.

**Summary**

This would centralize duplicated infrastructure across foundation-model wrappers, especially loading, caching, device handling, serialization, prediction formatting, quantization, PEFT, and pretraining.

pros:
- a lot of bloated code gets removed
- expanding capabilities will be easier
- interfacing new models will be easier
- end-to-end testing of each component can be ensured in a better way given we have a base model contract atleast
- improving inference/speed-ups and training will be easy at a bigger scale, doesn't need to be impelmentated for each estimator

cons:
- hard design and implementation problem
- not sure if the output is going to be very clean, how generalized the models can be
- main cost is that this is a difficult abstraction: current models are heterogeneous, including Hugging Face models, pipelines, GluonTS/Lightning models, JAX models, tokenizer/model pairs, and domain-specific forecasters

The refactor should be incremental, starting with a few representative models before migrating the full ecosystem.
